tests/main/layout-change/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: Ensure that replacing a bind with a symlink doesn't break an update
+
+details: |
+    Recently, the steam snap replaced several bindings with symlinks in the
+    layout part, and that resulted in users not being able to update it from
+    the "binded" version to the "symlinked" one because snapd complained that
+    it couldn't update the mount namespace. The solution consists on discard
+    the namespace and retry again.
+
+systems:
+    # the snaps aren't available for ARM or 32bit systems.
+    - -ubuntu-18.04-32
+    - -ubuntu-20.04-arm-64
+
+prepare: |
+    echo "Removing snaps"
+    snap remove --purge test-snapd-layout-change || true
+    snap remove --purge test-snapd-layout-change-with-daemon || true
+
+restore: |
+    echo "Removing snaps"
+    snap remove --purge test-snapd-layout-change || true
+    snap remove --purge test-snapd-layout-change-with-daemon || true
+
+execute: |
+    echo Testing layout change
```

Comment:
(nitpick) Sometimes echo uses "" and sometimes it does not, it does not really matter but for consistency it would be nice to keep using "" (or not but it should not switch in the middle :)

Reply:
Done.